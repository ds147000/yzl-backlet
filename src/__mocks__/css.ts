export default {
  process(): string {
    return '';
  },
};
