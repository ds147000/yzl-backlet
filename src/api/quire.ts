/* eslint-disable prettier/prettier */
/* eslint-disable import/prefer-default-export */
import request from '@/libs/api';

export type HashBlockDetailsResponse = {
  'hash'?: string;
  'ver'?: number;
  'prev_block'?: string;
  'mrkl_root'?: string;
  'time'?: number;
  'bits'?: number;
  'nonce'?: number;
  'n_tx'?: number;
  'size'?: number;
  'block_index'?: number;
  'main_chain'?: boolean;
  'height'?: number;
  'received_time'?: number;
  'relayed_by'?: string;
  'tx'?: HashBlockDetailsResponse[];
};

export async function GetHashBlockDetails(data: string) {
  return request.get<HashBlockDetailsResponse>(`https://blockchain.info/rawblock/${data}`);
}
