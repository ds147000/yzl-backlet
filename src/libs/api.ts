import axios, { AxiosRequestConfig, AxiosResponse, AxiosStatic, CancelTokenSource } from 'axios';

export class Request {
  private _axios: AxiosStatic;

  constructor(_axios: AxiosStatic) {
    this._axios = _axios;
    this.bindInterceptors();
  }

  bindInterceptors(): void {
    this._axios.interceptors.request.use(this.handleRequest, this.handleRequestError);
    this._axios.interceptors.response.use(this.handleResponse);
  }

  handleRequest(config: AxiosRequestConfig): AxiosRequestConfig {
    const prefix = window.blocklet ? window.blocklet.prefix : '/';
    config.baseURL = prefix || '';
    config.timeout = 200000;
    return config;
  }

  handleResponse(response: AxiosResponse<any>): any {
    return response.data;
  }

  async handleRequestError(error: any): Promise<any> {
    return Promise.reject(error);
  }

  async post<res>(url: string, data?: any): Promise<res> {
    return this._axios.post(url, data);
  }

  async get<res>(url: string, parmas?: any): Promise<res> {
    return this._axios.get(url, parmas);
  }

  getCloseSoure(): CancelTokenSource {
    return axios.CancelToken.source();
  }
}

const request = new Request(axios);

export default request;
