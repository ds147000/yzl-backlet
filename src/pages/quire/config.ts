/* eslint-disable import/prefer-default-export */
/* eslint-disable prettier/prettier */

export const LabelMap = {
  'hash': 'Hash',
  'ver': 'Ver',
  'prev_block': 'PrevBlock',
  'mrkl_root': 'Mrkl Root',
  'time': 'Time',
  'bits': 'Bits',
  'nonce': 'Nonce',
  'n_tx': 'N Text',
  'size': 'Size',
  'block_index': 'Block Index',
  'main_chain': 'Main Chain',
  'height': 'Height',
  'received_time': 'Received Time',
  'relayed_by': 'Relayed By',
  'tx': 'More',
};
