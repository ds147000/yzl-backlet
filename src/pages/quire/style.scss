.quire {
  .header {
    height: unit(100);
    background-color: white;
    box-shadow: 0 0 5px 0 $back;
  }

  .input {
    width: unit(280);
  }

  .list {
    position: relative;
    padding: unit(15) unit(20) unit(40) unit(20);
    box-sizing: border-box;
    width: 100vw;
    overflow: hidden;

    .mask {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      z-index: 5;
      background-color: $mask;
      box-sizing: border-box;
      padding-top: unit(150);
    }

    .item {
      border-bottom: 1px solid $border;
      padding: unit(10) unit(0);

      .label {
        width: unit(115);
        font-weight: 700;
        font-size: unit(14);
        color: $text2;
      }

      .value {
        color: $text;
        font-size: unit(12);
        word-wrap: break-word;
        word-break: break-all;
      }
    }
  }
}

@media screen and (max-width: 750px) {
  .quire {
    .ant-form .ant-form-item .ant-form-item-label {
      display: none;
    }

    .ant-form-inline .ant-form-item {
      margin-right: unit(0);
    }

    .list {
      .mask {
        padding-top: unit(200);
      }

      .item {
        .label {
          font-size: unit(12);
        }

        .value {
          font-size: unit(10);
          width: unit(250);
        }
      }
    }
  }
}
