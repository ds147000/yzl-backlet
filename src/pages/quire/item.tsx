import React from 'react';
import { HashBlockDetailsResponse } from '@/api/quire';
import { LabelMap } from './config';

type ItemValue = string | number | boolean | HashBlockDetailsResponse[];

interface ItemProps {
  label: keyof HashBlockDetailsResponse;
  value?: ItemValue;
}

const Item: React.FC<ItemProps> = ({ label, value }) => {
  return (
    <div className="item flex">
      <div className="label">{LabelMap[label]}</div>
      <div className="flex1 value" data-testid={label}>
        {valueToString(value)}
      </div>
    </div>
  );
};

function valueToString(value?: ItemValue): React.ReactNode {
  if (!value) return '';

  switch (typeof value) {
    case 'boolean':
      return value ? 'Yes' : 'No';

    case 'number':
      return String(value);

    case 'string':
      return value;

    default:
      return value.map((item) => (
        <p key={item.hash}>
          <a href={`/quire?hash=${item.hash}`} target={item.hash}>
            {item.hash}
          </a>
          ;
        </p>
      ));
  }
}

export default Item;
