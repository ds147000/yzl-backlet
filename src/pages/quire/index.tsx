import qs from 'qs';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { Form, Input, Button, message, Empty, Affix, Spin } from 'antd';
import { RouteComponentProps } from 'react-router-dom';
import { GetHashBlockDetails, HashBlockDetailsResponse } from '@/api/quire';
import { LabelMap } from './config';
import Item from './item';
import './style.scss';

const InputRules = [
  {
    required: true,
    message: '请输入 Block Hash',
    transform: (value: string) => String(value).trim(),
  },
];

type RouterPramas = {
  hash?: string;
};

const DelayTime = 100;

const Quire: React.FC<RouteComponentProps> = ({ history }) => {
  const [data, setData] = useState<HashBlockDetailsResponse>({});
  const [loading, setLoading] = useState(false);
  const keys = useMemo(() => Object.keys(LabelMap) as (keyof HashBlockDetailsResponse)[], [data]);
  const isEmpty = Object.values(data).length === 0;
  const params = qs.parse(history.location.search.slice(1)) as RouterPramas;

  const onSubmit = useCallback(
    ({ hash }) => {
      if (params.hash !== hash) history.push(`?hash=${hash}`);

      setLoading(true);
      GetHashBlockDetails(hash)
        .then((res) => setData(res))
        .catch((err) => {
          if (err.response.data?.message) message.error(err.response.data?.message);
        })
        .finally(() => setLoading(false));
    },
    [params.hash]
  );

  useEffect(() => {
    if (params.hash) onSubmit({ hash: params.hash });
  }, []);

  return (
    <div className="page quire">
      <Affix>
        <div className="header flex flex-ac flex-jc">
          <Form layout="inline" onFinish={onSubmit}>
            <Form.Item label="Block Hash" name="hash" rules={InputRules}>
              <Input placeholder="请输入Block Hash" className="input" />
            </Form.Item>
            <Form.Item>
              <Button type="primary" htmlType="submit" block>
                查询
              </Button>
            </Form.Item>
          </Form>
        </div>
      </Affix>
      <Spin spinning={loading} size="large" delay={DelayTime} tip="努力查询中...">
        <div className="list">
          {keys.map((key) => (
            <Item key={key} label={key} value={data[key]} />
          ))}
          {isEmpty && !loading && (
            <div className="mask">
              <Empty description="暂无数据" />
            </div>
          )}
        </div>
      </Spin>
    </div>
  );
};

export default Quire;
