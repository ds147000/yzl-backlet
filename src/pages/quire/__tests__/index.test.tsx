/* eslint-disable no-await-in-loop */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import { waitFor, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';
import { GetHashBlockDetails } from '@/api/quire';
import { GetHashBlockDetailsReuslt } from '@/api/__mocks__/quire';
import Quire from '../index';

jest.mock('@/api/quire');

const MockGetHashBlockDetails = GetHashBlockDetails as unknown as jest.Mock<typeof GetHashBlockDetails>;

beforeEach(() => {
  MockGetHashBlockDetails.mockClear();
});

describe('Quire', () => {
  const props: any = {
    history: {
      location: {
        search: '',
      },
      push: jest.fn(),
    },
  };

  test('空值查询', async () => {
    const Page = render(<Quire {...props} />);
    const subBtn = Page.getByTestId('submit');
    userEvent.click(subBtn);

    const keys = Object.keys(GetHashBlockDetailsReuslt);
    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(0));

    for (let i = 0; i < keys.length; i++) {
      const key = keys[i] as keyof typeof GetHashBlockDetailsReuslt;
      // eslint-disable-next-line no-await-in-loop
      await waitFor(() => expect(Page.getByTestId(key).innerHTML).toBe(''));
    }
  });

  test('输入空格查询', async () => {
    const Page = render(<Quire {...props} />);
    const subBtn = Page.getByTestId('submit');
    const input = Page.getByPlaceholderText('请输入Block Hash');

    userEvent.type(input, '       ');
    userEvent.click(subBtn);

    const keys = Object.keys(GetHashBlockDetailsReuslt);
    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(0));

    for (let i = 0; i < keys.length; i++) {
      const key = keys[i] as keyof typeof GetHashBlockDetailsReuslt;
      // eslint-disable-next-line no-await-in-loop
      await waitFor(() => expect(Page.getByTestId(key).innerHTML).toBe(''));
    }
  });

  test('输入值查询', async () => {
    const _props: any = {
      history: {
        location: {
          search: '',
        },
        push: jest.fn(),
      },
    };

    const Page = render(<Quire {..._props} />);
    const subBtn = Page.getByTestId('submit');
    const input = Page.getByPlaceholderText('请输入Block Hash');

    userEvent.type(input, 'abcsasakjklsajkajsklajkslwqwqsa');
    userEvent.click(subBtn);

    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(_props.history.push).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(_props.history.push.mock.calls[0][0]).toBe('?hash=abcsasakjklsajkajsklajkslwqwqsa'));

    const keys = Object.keys(GetHashBlockDetailsReuslt);
    for (let i = 0; i < keys.length; i++) {
      const key = keys[i] as keyof typeof GetHashBlockDetailsReuslt;
      const val = GetHashBlockDetailsReuslt[key];
      if (typeof val === 'string' || typeof val === 'number') {
        await waitFor(() => expect(Page.getByTestId(key).innerHTML).toBe(String(val)));
      } else if (val === true) {
        await waitFor(() => expect(Page.getByTestId(key).innerHTML).toBe('Yes'));
      } else if (val === false) {
        await waitFor(() => expect(Page.getByTestId(key).innerHTML).toBe('No'));
      }
    }
  });

  test('接口错误请客', async () => {
    const Page = render(<Quire {...props} />);
    const subBtn = Page.getByTestId('submit');
    const input = Page.getByPlaceholderText('请输入Block Hash');

    MockGetHashBlockDetails.mockRejectedValueOnce({
      response: {
        data: {
          message: 'server error',
        },
      },
    } as unknown as never);

    userEvent.type(input, 'abcsasakjklsajkajsklajkslwqwqsa');
    userEvent.click(subBtn);

    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(1));

    Page.getByText('暂无数据');
  });

  test('接口错误请客', async () => {
    const Page = render(<Quire {...props} />);
    const subBtn = Page.getByTestId('submit');
    const input = Page.getByPlaceholderText('请输入Block Hash');

    MockGetHashBlockDetails.mockRejectedValueOnce({
      response: {},
    } as unknown as never);

    userEvent.type(input, 'abcsasakjklsajkajsklajkslwqwqsa');
    userEvent.click(subBtn);

    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(1));

    Page.getByText('暂无数据');
  });

  test('刷新页面', async () => {
    const _props: any = {
      history: {
        location: {
          search: '?hash=1234567890',
        },
        push: jest.fn(),
      },
    };

    render(<Quire {..._props} />);
    await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(MockGetHashBlockDetails.mock.calls[0][0]).toBe('1234567890'));
  });

  test('重复搜索', async () => {
    const _props: any = {
      history: {
        location: {
          search: '',
        },
        push: jest.fn(),
      },
    };

    const mock = (hash: string) => {
      return new Promise((res) => {
        setTimeout(() => {
          if (hash === '123456789') res(GetHashBlockDetailsReuslt);
        }, 500);
      });
    };

    await act(async () => {
      const Page = render(<Quire {..._props} />);
      const subBtn = Page.getByTestId('submit');
      const input = Page.getByPlaceholderText('请输入Block Hash');

      MockGetHashBlockDetails.mockResolvedValueOnce(mock as unknown as never);
      userEvent.type(input, '6');
      userEvent.click(subBtn);
      await new Promise((res) => setTimeout(res, 400));

      MockGetHashBlockDetails.mockResolvedValueOnce(mock as unknown as never);
      userEvent.type(input, '789');
      userEvent.click(subBtn);

      await new Promise((res) => setTimeout(res, 800));

      await waitFor(() => expect(MockGetHashBlockDetails).toHaveBeenCalledTimes(2));
      await waitFor(() => expect(_props.history.push).toHaveBeenCalledTimes(2));
      await waitFor(() => expect(_props.history.push.mock.calls[0][0]).toBe('?hash=6'));
      await waitFor(() => expect(_props.history.push.mock.calls[1][0]).toBe('?hash=6789'));
    });
  });

  test('快照', async () => {
    const _props: any = {
      history: {
        location: {
          search: '?hash=1234567890',
        },
        push: jest.fn(),
      },
    };

    const Page = render(<Quire {..._props} />);
    await waitFor(() => expect(Page).toMatchSnapshot());
  });
});
