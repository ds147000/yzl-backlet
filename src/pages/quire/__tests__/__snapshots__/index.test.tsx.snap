// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quire 快照 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-message"
      >
        <div>
          <div
            class="ant-message-notice"
          >
            <div
              class="ant-message-notice-content"
            >
              <div
                class="ant-message-custom-content ant-message-error"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
                <span>
                  data is empty
                </span>
              </div>
            </div>
          </div>
          <div
            class="ant-message-notice"
          >
            <div
              class="ant-message-notice-content"
            >
              <div
                class="ant-message-custom-content ant-message-error"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    />
                  </svg>
                </span>
                <span>
                  server error
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <style
      rc-util-key="antd-wave"
    >
      
      [ant-click-animating-without-extra-node='true']::after, .ant-click-animating-node {
        --antd-wave-shadow-color: buttonface;
      }
    </style>
    <div>
      <div
        class="page quire"
      >
        <div>
          <div
            class=""
          >
            <div
              class="header flex flex-ac flex-jc"
            >
              <form
                class="ant-form ant-form-inline"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="hash"
                      title="Block Hash"
                    >
                      Block Hash
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input input"
                          id="hash"
                          placeholder="请输入Block Hash"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary ant-btn-block"
                          data-testid="submit"
                          type="submit"
                        >
                          <span>
                            查 询
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="list"
            >
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Hash
                </div>
                <div
                  class="flex1 value"
                  data-testid="hash"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Ver
                </div>
                <div
                  class="flex1 value"
                  data-testid="ver"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  PrevBlock
                </div>
                <div
                  class="flex1 value"
                  data-testid="prev_block"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Mrkl Root
                </div>
                <div
                  class="flex1 value"
                  data-testid="mrkl_root"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Time
                </div>
                <div
                  class="flex1 value"
                  data-testid="time"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Bits
                </div>
                <div
                  class="flex1 value"
                  data-testid="bits"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Nonce
                </div>
                <div
                  class="flex1 value"
                  data-testid="nonce"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  N Text
                </div>
                <div
                  class="flex1 value"
                  data-testid="n_tx"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Size
                </div>
                <div
                  class="flex1 value"
                  data-testid="size"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Block Index
                </div>
                <div
                  class="flex1 value"
                  data-testid="block_index"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Main Chain
                </div>
                <div
                  class="flex1 value"
                  data-testid="main_chain"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Height
                </div>
                <div
                  class="flex1 value"
                  data-testid="height"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Received Time
                </div>
                <div
                  class="flex1 value"
                  data-testid="received_time"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  Relayed By
                </div>
                <div
                  class="flex1 value"
                  data-testid="relayed_by"
                />
              </div>
              <div
                class="item flex"
              >
                <div
                  class="label"
                >
                  More
                </div>
                <div
                  class="flex1 value"
                  data-testid="tx"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page quire"
    >
      <div>
        <div
          class=""
        >
          <div
            class="header flex flex-ac flex-jc"
          >
            <form
              class="ant-form ant-form-inline"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="hash"
                    title="Block Hash"
                  >
                    Block Hash
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input input"
                        id="hash"
                        placeholder="请输入Block Hash"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn ant-btn-primary ant-btn-block"
                        data-testid="submit"
                        type="submit"
                      >
                        <span>
                          查 询
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="list"
          >
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Hash
              </div>
              <div
                class="flex1 value"
                data-testid="hash"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Ver
              </div>
              <div
                class="flex1 value"
                data-testid="ver"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                PrevBlock
              </div>
              <div
                class="flex1 value"
                data-testid="prev_block"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Mrkl Root
              </div>
              <div
                class="flex1 value"
                data-testid="mrkl_root"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Time
              </div>
              <div
                class="flex1 value"
                data-testid="time"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Bits
              </div>
              <div
                class="flex1 value"
                data-testid="bits"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Nonce
              </div>
              <div
                class="flex1 value"
                data-testid="nonce"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                N Text
              </div>
              <div
                class="flex1 value"
                data-testid="n_tx"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Size
              </div>
              <div
                class="flex1 value"
                data-testid="size"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Block Index
              </div>
              <div
                class="flex1 value"
                data-testid="block_index"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Main Chain
              </div>
              <div
                class="flex1 value"
                data-testid="main_chain"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Height
              </div>
              <div
                class="flex1 value"
                data-testid="height"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Received Time
              </div>
              <div
                class="flex1 value"
                data-testid="received_time"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                Relayed By
              </div>
              <div
                class="flex1 value"
                data-testid="relayed_by"
              />
            </div>
            <div
              class="item flex"
            >
              <div
                class="label"
              >
                More
              </div>
              <div
                class="flex1 value"
                data-testid="tx"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
